Cargo.toml

```diff
@@ -8,21 +8,25 @@ license = "MIT"
 homepage = "https://github.com/snapview/tokio-tungstenite"
 documentation = "https://docs.rs/tokio-tungstenite/0.9.0"
 repository = "https://github.com/snapview/tokio-tungstenite"
-version = "0.9.0"
+version = "0.10.0-alpha.1"
 edition = "2018"
 
 [features]
 default = ["connect", "tls"]
-connect = ["tokio-dns-unofficial", "tokio-tcp", "stream"]
+connect = ["tokio-net", "stream"]
 tls = ["tokio-tls", "native-tls", "stream", "tungstenite/tls"]
 stream = ["bytes"]
 
 [dependencies]
-futures = "0.1.23"
-tokio-io = "0.1.7"
+log = "0.4"
+futures = { version = "0.3", features = ["async-await"] }
```

Comment:
This will break things until tokio is using futures-preview

Reply:
Related tokio issue: https://github.com/tokio-rs/tokio/issues/1708 There should be a new alpha release soon with the new stable futures as dependencies.

Weirdly, I'm currently using this updated pull request in a project together with the current tokio alpha (the futures-preview based one) and it seems to work.

Reply:
Tokio master branch is already using it, so it is just a matter of a new release. Also, this change did not break things for me :man_shrugging:

Reply:
The `async-await` feature is enabled by default in futures 0.3, so no need to explicitly enable it anymore.

Reply:
I think it would make sense to release the alpha version of the crate, as `hyper` and `reqwest` do. In this case, it's better to use fixed versions of `futures-preview` and `tokio` for now.

Reply:
We're still waiting on a release of tungstenite-rs before an alpha of this can be released.

Reply:
> We're still waiting on a release of tungstenite-rs before an alpha of this can be released.

`tungstenite-rs` 0.9.2 has been published today.